cmd/soroban-rpc/internal/methods/simulate_transaction.go

```diff
@@ -67,10 +67,19 @@ func NewSimulateTransactionHandler(logger *log.Entry, ledgerEntryReader db.Ledge
 			sourceAccount = txEnvelope.SourceAccount().ToAccountId()
 		}
 
-		xdrOp, ok := op.Body.GetInvokeHostFunctionOp()
-		if !ok {
+		footprint := xdr.LedgerFootprint{}
+		switch op.Body.Type {
+		case xdr.OperationTypeInvokeHostFunction:
+		case xdr.OperationTypeBumpFootprintExpiration, xdr.OperationTypeRestoreFootprint:
+			if txEnvelope.Type != xdr.EnvelopeTypeEnvelopeTypeTx && txEnvelope.V1.Tx.Ext.V != 1 {
+				return SimulateTransactionResponse{
+					Error: "Footprint operation provided without SorobanTransactionData",
```

Comment:
It would be better to have more explicit error message:
```To perform a SimulateTransaction for BumpFootprintExpiration or RestoreFootprint operations, SorobanTransactionData must be provided```